### ISSUE TITLE
about the weight of consistency losses

### ISSUE THREAD

**Issue by henniekim** [ZHKKKe/DualStudent#1]
Hi !
I like your idea and I want to use your code as a baseline to study SSL. 
However, I wonder that the scale of consistency losses.
You set the weights to 1:10:100. 
(Plus, I've also found that 'mean teacher' which is your baseline, set the consistency weight to 100. )
I think these kind of values are not general since people usually set weight equally. 

In my opinion, small weight for 'supervised loss' can help prevent from overfitting but it seems that it is too immoderate to choose such 1:100. 

I'd like to ask why you set these parameters as such values.

Thanks !

**Comment by ZHKKKe** [comment]
Hi, Thanks for your attention!

There are two consistency terms in our method:
(1) The consistency loss inside each student, which is weighted by the 'consistency_scale' in the script.
(2) The stabilization loss between two students, which is weighted by the 'stabilization_scale' in the script.

As analyzed in the ablation study section (Fig. 7 [right]), the stabilization loss (L_sta) is the main consistency term in our method. The consistency loss inside each student (L_con) can only improve the performance slightly. In practice, set 'consistency_scale' to 100 will get the almost same performance since our method is not sensitive to this hyperparameter. In our script, we just set the weights to 1:10:100 to highlight the stabilization loss.

In the Mean Teacher and our method (or other consistency-based semi-supervised classification methods), the ‘supervised loss’ uses CrossEntropy while the ‘consistency loss’ uses MSE (or KL-Divergence). We should use 1:100 to weight them due to the different loss types. If you change MSE to KL-Divergence, you should adjust their weights. (The target of the 'consistency loss' is not a one-hot tensor, that is why we do not use CrossEntropy for the 'consistency loss'.)
